lib/compress/zstdmt_compress.c

```diff
@@ -1261,7 +1261,9 @@ static size_t ZSTDMT_compress_advanced_internal(
     unsigned const compressWithinDst = (dstCapacity >= ZSTD_compressBound(srcSize)) ? nbJobs : (unsigned)(dstCapacity / ZSTD_compressBound(avgJobSize));  /* presumes avgJobSize >= 256 KB, which should be the case */
     size_t frameStartPos = 0, dstBufferPos = 0;
     assert(jobParams.nbWorkers == 0);
+#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
```

Comment:
Look 3 lines above (line 1261). I don't know what exactly `compressWithinDst` is used for off the top of my head, but the comment implies that it may need to be adjusted.

Reply:
`compressWithinDst` means that the compress output will be written directly into `dst`, instead of using some additional intermediate buffer.
In which case, it is indeed extremely important that `avgJobSize >= 256 KB`.
This is required so that `sum(ZSTD_compressBound(jobSize)) <= ZSTD_compressBound(inputSize)`.
If this condition is not respected, some output could be written out of bound.